### DISCUSSION TITLE
Borg server and client compatibility

### DISCUSSION THREAD

**Discussion by oixdbb** [borgbackup/borg#7970] [Q&A] [answered]
Will a server using borg 1.2 be able to handle borg clients using 1.1 repos?

**Comment by ThomasWaldmann** [comment, marked as answer]
Yes, just read the hints in the changelog.

**Comment by mauromol** [comment]
@ThomasWaldmann can you please clarify whether the upgrade notes described in documentation (https://borgbackup.readthedocs.io/en/stable/changes.html#upgrade-notes), when they are generically talking about "upgrading borg", are meant to be understood as "upgrading borg *client*" or rather *server*?

I mean. I currently have a bog 1.1.15 client writing to a server which was originally using borg 1.1.x itself. With time, the server was upgraded up to 1.2.6, and now I have a proposed upgrade to 1.4.x. (not yet done, though).
I *never* followed any upgrade guide on this repo, yet, I never had any problems creating, pruning and checking backups (*always" with the same 1.1.15 client). So, I was now wondering myself if I should start to follow some "upgrade guides" or if I should do that only whenever I will upgrade the client version.

Thank you!

**Reply by ThomasWaldmann** [reply]
Borg does most stuff client-side and we try to keep the server-side code compatible to older clients (as far as borg 1.x is concerned).

So I think a borg 1.4.0 server should work with borg 1.1.x clients.

Due to the CVE listed at top of the changelog, one must not mix borg client versions when they access the same repo(s) crossing the 1.2.4/1.2.5 version boundary. Either have all clients < 1.2.5  or have all clients >= 1.2.5 (and follow the upgrade notes).

**Reply by mauromol** [reply]
> Borg does most stuff client-side and we try to keep the server-side code compatible to older clients (as far as borg 1.x is concerned).
> 
> So I think a borg 1.4.0 server should work with borg 1.1.x clients.

Thank you Thomas, I would just suggest you to add this (IMHO important) clarification on the Upgrade Notes section of the documentation. That is that, when you talk about borg upgrades, you're actually talking about borg _client_ upgrades.
